src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,27 @@
 
     @Override
     public Country add(Country country) {
-        return null;
+        Transaction transaction = null;
+        try (Session session = factory.openSession()) {
+            transaction = session.beginTransaction();
+            session.save(country);
+            transaction.commit();
+            return country;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Error adding country", e);
+        }
     }
 
     @Override
     public Optional<Country> get(Long id) {
-        return null;
+        try (Session session = factory.openSession()) {
+            Country country = session.get(Country.class, id);
+            return Optional.ofNullable(country);
+        } catch (Exception e) {
+            throw new DataProcessingException("Error retrieving country", e);
```

Comment:
For the `get` method, consider adding more context to the exception message by including the country ID that caused the error. This can be helpful for debugging purposes.